### DISCUSSION TITLE
Is it possible to delete a channel from a lineup

### DISCUSSION THREAD

**Discussion by xjim1** [OpenSageTV/Community#171] [General]
Some time ago, a local station on channel 12 re-arranged it's sub-channel lineup, adding a new sub in the process, 12.7.  It also moved the content/network which had been on 12.3 to 12.7.  Sage never picked up this new sub, even after initiating a new scan, so I manually added it, giving it a station and description that seemed to make sense.  

The station tuned just fine, and with the expected network, but never picked up any EPG data further than 12 or so hours out.  It also lacked any episode or description information.  Followed up with Schedules Direct, and they told me that the broadcaster had to provide data for that new sub to Gracenote, etc, etc.  Long story short, I still don't see expected guide data and just learned to live with it.

Then I read this post:
[https://github.com/OpenSageTV/Community/issues/102](url)
and got me to thinking I need to provide more specific station info.  

I've done a rescan, and it just recognizes the channel I manually added originally, and I can't seem to update any station or ID information.
Is there a way to delete this subchannel, so Sage can rescan and maybe 12.7 will get picked up and id'd correctly?

Thanks for any help.

**Comment by JustFred51** [comment]
The fact that you're only seeing a few hrs of EPG data indicates that Sage is pulling this from the program stream itself, NOT from Schedules Direct. 

If the tuner in question is an HDHomeRun tuner, remember to use the HDHR Setup program to perform a channel scan BEFORE you do the Sage channel scan.  Verify that HdHr Setup detected the new/missing sub-stream.  (Some folks like to do channel remapping within HdHr Setup, but I prefer to remap within Sage itself).

Regardless of tuner type, in Sage, go into Channel Setup screen for the affected tuner.  Scroll down and locate the channel of interest in the left-hand list.  Select it and in the right-hand side of the screen, verify that the Station ID # is greater than 10000.  Smaller ID values aren't valid for Schedules Direct, and will indicate "No EPG data available".  If this is the case, follow the steps in the SageTv Manual to "remap" that channel.  Remember to do a Sage scan for every tuner using a given lineup.  I find it's less work to do the remapping just once after having scanned all tuners on that lineup.

**Comment by xjim1** [comment]
That's just it, the only thing I can remap is the logical or physical chan #'s.  The station id is 2, but I don't see a way to change that parameter, and I don't know what the station id would be even if I could.  

I'm using a Hauppauge 4 tuner card.

**Reply by JustFred51** [reply]
See page 173 of the SageTv Manual for Remap Station.

**Comment by xjim1** [comment]
I must have read through that part a half dozen times.  Thanks for the help and your patience.   I deleted the sub in question, and after a rescan, the channel came up, but w/a station id of 4.  I may be back to square one, but at least the channel was seen this time.  

So, with an id that low, is this something passed straight through from the broadcast signal, or could Sage be getting in the way?
All the other subchannels for that channel(7 of them) have id's above 10000.

I consider my initial problem solved, but still puzzled why this station is an outlier.  I'll wait for an EPG scan, but I'm not expecting it to pick up a full schedule.

**Comment by xjim1** [comment]
A quick look at Zap2it's EPG shows it does not list the channel in question.
Schedules Direct had always said the problem was between the broadcaster and Gracenote.  Don't know the role those companies play in getting epg data for my channel.

**Comment by JustFred51** [comment]
Good to hear you're making some progress.  But as long as the station ID # is less than 10000, there won't be EPG data for that station, no mater how many EPG updates occur.  Obviously, SD is correct in stating the problem is between the broadcaster and Gracenote.

But you might still be able to overcome the situation.  Go again to Remap Channel screen; scroll thru the (usually long) Remap Station list on the right side of the screen.  My preference is to select Options, then set Sorting Method to Sort By Name, then scroll thru the list looking for a Station Name that carries the program feed you're getting (with a Station ID greater than 10000), and select it to Remap the Station.  

I have a similar situation.  My cable company (which uses ClearQAM) mangles almost *all* Station Names.  For example, Sage scans as "KABC" but Schedules Direct knows this as "KABC-DT", requiring me to manually remap KABC to KABC-DT.   EPG data is fine after the remap.

**Comment by xjim1** [comment]
I scanned through the list of channels(a couple times) and there is another station, but I think it was the one I deleted earlier as it has an id of 2.

If the station id is not in the range, isn't that on the broadcaster, and not any intermediary?

**Comment by JustFred51** [comment]
Since neither Schedules Direct nor Zap2It's EPG lists the station name in question, I strongly suspect the broadcaster isn't providing that lineup info to the provider of the data (looks like SD gets it from Gracenote/DataDirect these days).  Until that occurs, SD isn't going to have a valid Station ID for that lineup.   Probably the best recourse is to report the missing lineup to SD at https://www.schedulesdirect.org/lineupsupport

**Comment by xjim1** [comment]
I'm going to contact SD again, but not expecting much.  My attempt at communicating with the local station was fruitless.
Is there maybe a plugin where I could change the station id directly?  There's a station in another market which shows up in my channel list, but I can't receive.  Maybe I could just remap the id and fool it into thinking it's that channel's epg data.

I scanned through all the plugins, and got a little blurry eyed by the end, but I found this one:
Schedules Direct EPG Plugin (OSS)
I briefly read through the user guide here:
[https://github.com/Slugger/sdepg](url)

It looks like a drastic action, and I can't tell if it will even let me do what I want.
Any idea welcome.

**Comment by JustFred51** [comment]
Selecting the station from another market should work.  Keep the Station ID and name, but remap the Logical and Physical channel #s.

I'm pretty sure the SD plugin on GitHub is a very old version before SD was integrated into Sage.  I'd advise against installing it, and doubt it'll have any more capability than the current built-in one does.

**Comment by xjim1** [comment]
I think that did it.  I was so preoccupied with changing the station id I never thought about doing it the other way around.  It tunes into the target channel, the epg finally shows correct programming out several days and I tuned my favorites to the new channel.  I'll keep the original subchannel for now.
Thank you for the help!!
